packages/hash/frontend/next.config.js

```diff
@@ -52,33 +52,70 @@ process.env.NEXT_PUBLIC_SENTRY_REPLAY_SESSION_SAMPLE_RATE =
-
-      // We call linters in GitHub Actions for all pull requests. By not linting
-      // again during `next build`, we save CI minutes and unlock more feedback.
-      // Thus, we can get Playwright test results and Preview releases for WIP PRs.
-      eslint: { ignoreDuringBuilds: true },
-      typescript: { ignoreBuildErrors: true },
-
-      sentry: {
-        autoInstrumentServerFunctions: false,
-        hideSourceMaps: false,
-      },
+    withTM(
+      /** @type {import('next').NextConfig} */
+      {
+        pageExtensions: [
+          "page.tsx",
+          "page.ts",
+          "page.jsx",
+          "page.jsx",
+          "api.ts",
+        ],
 
-      experimental: {
-        allowMiddlewareResponseBody: true,
-      },
+        // We call linters in GitHub Actions for all pull requests. By not linting
+        // again during `next build`, we save CI minutes and unlock more feedback.
+        // Thus, we can get Playwright test results and Preview releases for WIP PRs.
+        eslint: { ignoreDuringBuilds: true },
+        typescript: { ignoreBuildErrors: true },
+
+        sentry: {
+          autoInstrumentServerFunctions: false,
+          hideSourceMaps: false,
+        },
+
+        experimental: {
+          allowMiddlewareResponseBody: true,
+        },
+
+        webpack: (webpackConfig, { isServer }) => {
+          webpackConfig.module.rules.push({
+            test: /\.svg$/,
+            use: [require.resolve("@svgr/webpack")],
+          });
+
+          // eslint-disable-next-line no-param-reassign
+          webpackConfig.experiments.asyncWebAssembly = true;
+          if (!isServer) {
+            // eslint-disable-next-line no-param-reassign
+            webpackConfig.output.publicPath = `/_next/`;
+          } else {
+            // eslint-disable-next-line no-param-reassign
+            webpackConfig.output.publicPath = `./`;
+          }
+          // eslint-disable-next-line no-param-reassign
+          webpackConfig.output.assetModuleFilename = `static/[hash][ext]`;
+          webpackConfig.module.rules.push({
+            test: /\.(wasm)$/,
+            type: "asset/resource",
+          });
+
+          // eslint-disable-next-line no-param-reassign
+          webpackConfig.resolve.alias["@blockprotocol/type-system$"] =
+            "@blockprotocol/type-system/slim";
 
-      webpack: (webpackConfig) => {
-        webpackConfig.module.rules.push({
-          test: /\.svg$/,
-          use: [require.resolve("@svgr/webpack")],
-        });
+          // WebStorm doesn't understand exports so we have to use the full import path,
+          // but we can't because it's not in the exports so webpack complains
+          // This fixes that
+          // eslint-disable-next-line no-param-reassign
+          webpackConfig.resolve.alias[
+            "@blockprotocol/type-system/dist/wasm/type-system.wasm"
+          ] = require.resolve("@blockprotocol/type-system/type-system.wasm");
```

Comment:
This also feels uncomfortable to me. Are you sure it's necessary? 

I've just opened your branch in CLion (which I'd expect to be largely the same as WebStorm) and I'm not seeing an error when I delete this.

Reply:
Actually I can recreate the error after rewriting the import. I'm not sure why this is happening, when it's not happening when I open the demo. I'm wondering if it's to do with the [custom.d.ts](https://github.com/Alfred-Mountfield/wasm-sandbox/blob/main/nextjs-app/custom.d.ts) or something else

Reply:
We need to do two things to fix the errors I believe. 
1. We need to add an exclusion for the type system here
https://github.com/hashintel/hash/blob/9723f5f012b0b75f8797a4a6966b1ad2f04095dd/libs/javascript/%40local/eslint-config/legacy-base-eslintrc-to-refactor.cjs#L65-L74

2. We need to add the custom declaration (@kachkaev recommends we do it in https://github.com/hashintel/hash/blob/main/packages/hash/frontend/assets.d.ts) like I have in the `custom.d.ts` linked

Reply:
Jumped on a call to pair on this, and pushed a commit